src/components/widgets/Form.js

```diff
@@ -0,0 +1,315 @@
+          </div>
+        </form>
+        <div className="widget-form-preview">
+          <Preview image={image} description={description}/>
+        </div>
+      </div>
+    )
+  }
+}
+
+class Section extends Component {
+  render() {
+    const { type, fields, setFieldState, state } = this.props
+
+    return (
+      <div className="widget-form-section">
+        <div className="widget-form-section-type">
+          {type === 'Setup'
+            ? 'Setup — Now set up your widget'
+            : 'Source — Where is your data coming from?'
+          }
+        </div>
+        <div className="widget-form-section-fields">
+          {fields.map(( field, id ) => (
+              <Row
+                {...field}
+                setFieldState={setFieldState}
+                state={state}
+                key={`field-${id}`}
+              />
+            )
+          )}
+        </div>
+      </div>
+    )
+  }
+}
+
+class Preview extends Component {
+  render() {
+    const { description, image, title } = this.props
+    return (
+      <div className="widget-form-preview">
+        <img alt={title} src={image} />
+        <div>{description}</div>
+      </div>
+    )
+  }
+}
+
+class Field extends Component {
+  render() {
+    const { type } = this.props
+
+    if( type === 'text' ) {
+      return <Text {...this.props} />
+    } else if( type === 'dropdown' ) {
+      return <Dropdown {...this.props} />
+    } else if( type === 'nested-dropdown' ) {
+      return <NestedDropdownArea {...this.props} />
+    } else if( type === 'radio' ) {
+      return <RadioField {...this.props} />
+    } else {
+      return <div>Not a valid component type >_<</div>
+    }
+  }
+}
+
+class Row extends Component {
+  render () {
+    const { name, type } = this.props
+    if ( type === 'nested-dropdown' ) {
+      return (
+        <div className="widget-form-field">
+          <Field {...this.props} />
+        </div>
+      )
+    } else {
+      return (
+        <div className="widget-form-field">
+          <label>{name}</label>
+          <Field {...this.props} />
+        </div>
+      )
+    }
+  }
+}
+
+class Text extends Component {
+  constructor ( props ) {
+    super( props )
+    this.onChangeHandler = this.onChangeHandler.bind( this )
+  }
+
+  onChangeHandler( event ) {
+    const { setFieldState, name } = this.props
+    setFieldState( name, event.target.value )
+  }
+
+  render() {
+    const { state, name } = this.props
+    return <input
+      type="text"
+      defaultValue={state[name]}
+      onChange={this.onChangeHandler} />
+  }
+}
+
+class Dropdown extends Component {
+  constructor ( props ) {
+    super( props )
+    this.onChangeHandler = this.onChangeHandler.bind( this )
+  }
+
+  onChangeHandler( event ) {
+    const { setFieldState, name } = this.props
+    setFieldState( name, event.target.value )
+  }
+
+  render() {
+      const { options, name, state } = this.props
+      let dropdownOptions = []
```

Comment:
Instead of using a let here, I would move this into its own function:
```jsx
options() {
  return this.props.options.map( option => 
    <option value={option.value} key={`etc`}>{option.label}</option> 
  )
}
```